exasol_advanced_analytics_framework/event_handler/context/top_level_event_handler_context.py

```diff
@@ -0,0 +1,219 @@
+
+    def get_child_event_handler_context(self) -> ScopeEventHandlerContext:
+        self._check_if_valid()
+        temporary_path = self.get_temporary_path()
+        new_temporary_bucketfs_location = self._temporary_bucketfs_location.joinpath(temporary_path)
+        child_event_handler_conext = _ChildEventHandlerContext(
+            self,
+            new_temporary_bucketfs_location,
+            self._get_temporary_db_object_name(),
+            self._temporary_schema_name
+        )
+        self._child_event_handler_context_list.append(child_event_handler_conext)
+        return child_event_handler_conext
+
+    def _is_child(self, scope_event_handler_context: ScopeEventHandlerContext):
+        result = isinstance(scope_event_handler_context, _ChildEventHandlerContext) and \
+                 scope_event_handler_context._parent == self
+        return result
+
+    def _transfer_object_to(self, object_proxy: ObjectProxy,
+                            scope_event_handler_conext: ScopeEventHandlerContext):
+        self._check_if_valid()
+        if object_proxy in self._owned_object_proxies:
+            scope_event_handler_conext._own_object(object_proxy)
+            if not self._is_child(scope_event_handler_conext):
+                self._remove_object(object_proxy)
+        else:
+            raise RuntimeError("Object not owned by this ScopeEventHandlerContext.")
+
+    def _remove_object(self, object_proxy: ObjectProxy):
+        self._valid_object_proxies.remove(object_proxy)
+        self._owned_object_proxies.remove(object_proxy)
+
+    def _check_if_valid(self):
+        if not self._is_valid:
+            raise RuntimeError("Context already released.")
+
+    def _invalidate(self):
+        self._check_if_valid()
+        self._invalid_object_proxies = self._invalid_object_proxies.union(self._valid_object_proxies)
+        self._valid_object_proxies = set()
+        self._owned_object_proxies = set()
+        self._is_valid = False
+        for child_event_handler_conext in self._child_event_handler_context_list:
+            child_event_handler_conext._invalidate()
+
+
+class TopLevelEventHandlerContext(_ScopeEventHandlerContextBase):
+    def __init__(self,
+                 temporary_bucketfs_location: AbstractBucketFSLocation,
+                 temporary_db_object_name_prefix: str,
+                 temporary_schema_name: str):
+        super().__init__(temporary_bucketfs_location,
+                         temporary_db_object_name_prefix,
+                         temporary_schema_name)
+
+    def _release_object(self, object_proxy: ObjectProxy):
+        self._check_if_valid()
+        self._valid_object_proxies.remove(object_proxy)
+        if object_proxy in self._owned_object_proxies:
+            self._owned_object_proxies.remove(object_proxy)
+        self._invalid_object_proxies.add(object_proxy)
+        object_proxy._invalidate()
+
+    def _register_object(self, object_proxy: ObjectProxy):
```

Comment:
I would implement the  _register_object() method in parent class, because the task of adding the object into valid_proxies should be same for all subclasses.

Reply:
so you mean we extract it to the parent as is

Reply:
yes exactly